applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+        # method, it again spreads condition NORMALs to nodes where condition is applied
+        # with SLIP flag.
+        RansVariableUtilities.AssignConditionVariableValuesToNodes(
+            model_part, Kratos.NORMAL, Kratos.SLIP)
+
+        Kratos.Logger.PrintInfo("NormalCalculationUtils",
+                                "Condition normals calculated.")
+
+
+def CreateRansFormulationModelPart(formulation, element_name, condition_name):
+    formulation.domain_size = formulation.GetBaseModelPart().ProcessInfo[
+        Kratos.DOMAIN_SIZE]
+
+    element_suffix = str(
+        formulation.domain_size) + "D" + str(formulation.domain_size + 1) + "N"
+    condition_suffix = str(formulation.domain_size) + "D" + str(
+        formulation.domain_size) + "N"
+
+    element_name = element_name + element_suffix
+    condition_name = condition_name + condition_suffix
+    return CreateDuplicateModelPart(
+        formulation.GetBaseModelPart(),
+        formulation.GetName() + "_" + element_name + "_" + condition_name,
+        element_name, condition_name, "")
+
+
+def IsBufferInitialized(formulation):
+    return (formulation.GetBaseModelPart().ProcessInfo[Kratos.STEP] + 1 >=
+            formulation.GetMinimumBufferSize())
+
+
+def InitializePeriodicConditions(base_model_part, model_part, variables_list):
+    properties = model_part.CreateNewProperties(
+        model_part.NumberOfProperties() + 1)
+    pcu = KratosCFD.PeriodicConditionUtilities(
+        model_part, model_part.ProcessInfo[Kratos.DOMAIN_SIZE])
+    for variable in variables_list:
+        pcu.AddPeriodicVariable(properties, variable)
+
+    index = model_part.NumberOfConditions()
+    for condition in base_model_part.Conditions:
+        if condition.Is(Kratos.PERIODIC):
+            index += 1
+            node_id_list = [node.Id for node in condition.GetNodes()]
+            periodic_condition = model_part.CreateNewCondition(
+                "PeriodicCondition", index, node_id_list, properties)
+            periodic_condition.Set(Kratos.PERIODIC)
+
+
+def CreateSteadyScalarScheme(relaxation_factor):
+    return steady_scalar_scheme(relaxation_factor)
+
+
+def CreateSteadyAlgebraicFluxCorrectedScheme(relaxation_factor,
+                                                     boundary_flags,
+                                                     is_periodic):
+    if (is_periodic):
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags,
+                                        KratosCFD.PATCH_INDEX)
+    else:
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags)
+
+
+def CreateBossakRelaxationScalarScheme(bossak_value, relaxation_factor, scalar_variable,
+                             scalar_rate_variable,
+                             relaxed_scalar_rate_variable):
+    return bossak_scheme(bossak_value, relaxation_factor, scalar_variable,
+                         scalar_rate_variable, relaxed_scalar_rate_variable)
```

Comment:
Same, I think this should be encapsulated within the solver.

Reply:
removed :)